src/govuk/components/notification-banner/notification-banner.yaml

```diff
@@ -0,0 +1,177 @@
+params:
+- name: text
+  type: string
+  required: true
+  description: If `html` is set, this is not required. Text to use within the notification banner. If `html` is provided, the `text` argument will be ignored.
+- name: html
+  type: string
+  required: true
+  description: If `text` is set, this is not required. HTML to use within the notification banner. If `html` is provided, the `text` argument will be ignored.
+- name: title
+  type: string
+  required: false
+  description: Title text to use within the notification banner. Defaults to 'Important' ('Success' for success type and 'Error' for error type). If `titleHtml` is supplied, the `title` argument will be ignored.
+- name: titleHtml
+  type: string
+  required: false
+  description: Title HTML to use within the notification banner. If `titleHtml` is provided, the `title` argument will be ignored.
+- name: titleHeadingLevel
+  type: string
+  required: false
+  description: Heading level, from 1 to 6. Default is `2`.
+- name: type
+  type: string
+  required: false
+  description: If `type` is set to `success` or `error`, the notification banner sets `role` to `alert` and `tabindex` to `-1`, and JavaScript moves the keyboard focus moves to the notification banner when the page loads. If `type` is not set, the notification banner defaults to setting `role` to `region` and using `aria-labelledby` to provide information for users of assistive technologies.
+- name: role
+  type: string
+  required: false
+  description: Overrides the value of the `role` attribute for the notification banner. Defaults to `region`. If `type` is set to `success` or `error`, defaults to `alert`.
+- name: tabindex
+  type: string/boolean
+  required: false
+  description: Overrides the value of the `tabindex` attribute for the notification banner. Not set by default.  If `type` is set to `success` or `error`, defaults to `-1`; the attribute can be removed by setting `tabindex` to `false`;
```

Comment:
The description for the type option above mentions the tabindex attribute, but doesn’t explain it. Would it help users if this tabindex description displayed before they read about the type option?

Are options set by default unless we state otherwise? If so, do we need to call this out somewhere?

Instead of "Not set by default," could we reword the first sentence to "Overrides the value that the browser sets for the tabindex attribute’ (or similar)?

“If type is set to success or error, defaults to alert.” Is it the tabindex or the type that defaults?

Suggested tweak for last sentence: “If you set type to success or error, tabindex defaults to -1. To remove the attribute, set tabindex to false.”